GoMeet/src/test/java/com/google/sps/PopularLocationServletTest.java

```diff
@@ -0,0 +1,149 @@
+    servlet.setDao(mockedLocationDao);
+    servlet.doGet(request, response);
+
+    // Check if the popular location was sent in the response
+    stringWriter.flush();
+    Type locationListType = new TypeToken<ArrayList<Location>>(){}.getType();
+    ArrayList<Location> locationList = gson.fromJson(stringWriter.toString(), locationListType);
+
+    assertTrue(locationList.contains(LOCATION_B));
+    assertFalse(locationList.contains(LOCATION_A));
+  }
+
+  /**
+   * Tests if the response contains the most popular locations when there are 
+   * multiple popular locations.
+   */
+  @Test
+  public void doGetTestMultiple() throws IOException {
+    // Set up DAO mock
+    List<Location> listToReturn =
+        new ArrayList<>(Arrays.asList(LOCATION_A, LOCATION_B, LOCATION_C));
+    when(mockedLocationDao.getAll()).thenReturn(listToReturn);
+
+    PopularLocationServlet servlet = new PopularLocationServlet();
+    servlet.setDao(mockedLocationDao);
+    servlet.doGet(request, response);
+
+    // Check if the popular locations were sent in the response
+    stringWriter.flush();
+    Type locationListType = new TypeToken<ArrayList<Location>>(){}.getType();
+    ArrayList<Location> locationList = gson.fromJson(stringWriter.toString(), locationListType);
+
+    assertTrue(locationList.contains(LOCATION_B));
+    assertTrue(locationList.contains(LOCATION_C));
+    assertFalse(locationList.contains(LOCATION_A));   
+  }
+
+  /** Tests that an empty array is sent when there are no locations in the database. */
+  @Test
+  public void doGetEmptyDatabase() throws IOException {
+    List<Location> emptyList = new ArrayList<>();
+    when(mockedLocationDao.getAll()).thenReturn(emptyList);
+
+    PopularLocationServlet servlet = new PopularLocationServlet();
+    servlet.setDao(mockedLocationDao);
+    servlet.doGet(request, response);
+
+    // Check if the no locations were sent in the response
+    stringWriter.flush();
+    Type locationListType = new TypeToken<ArrayList<Location>>(){}.getType();
+    ArrayList<Location> locationList = gson.fromJson(stringWriter.toString(), locationListType);
+
+   assertTrue(locationList.isEmpty());
+  }
+
+  /** 
+   * Tests that an array with all the locations is sent when the locations'
+   * vote counts are the same.
+   */
+  @Test
+  public void doGetSameVoteCount() throws IOException {
+    // Set up DAO mock
+    List<Location> listToReturn = new ArrayList<>(Arrays.asList(LOCATION_B, LOCATION_C));
+    when(mockedLocationDao.getAll()).thenReturn(listToReturn);
+
+    PopularLocationServlet servlet = new PopularLocationServlet();
+    servlet.setDao(mockedLocationDao);
+    servlet.doGet(request, response);
+
+    // Check if the popular location was sent in the response
+    stringWriter.flush();
+    Type locationListType = new TypeToken<ArrayList<Location>>(){}.getType();
+    ArrayList<Location> locationList = gson.fromJson(stringWriter.toString(), locationListType);
```

Comment:
optional - can extract these lines into a helper function to avoid repeated code

Reply:
Yup, I've decided to set up the servlet with the mockedDao is the @Before setup to remove the repeated code :)